[quant][qat] Ensure observers and fq modules are scriptable

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #44773 [quant][qat] Ensure fake_quant and observer can be disabled on scriptmodule
* #44765 [quant][fx] Add node name as prefix to observer module name
* **#44749 [quant][qat] Ensure observers and fq modules are scriptable**

Summary:
Ensure fx module is scriptable after calling prepare_qat on it

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_qat_and_script

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23718380](https://our.internmc.facebook.com/intern/diff/D23718380)

supriyar committed:
[quant][qat] Ensure observers and fq modules are scriptable

Summary:
Ensure fx module is scriptable after calling prepare_qat on it

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_qat_and_script

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/quantization/fake_quantize.py:
```diff
@@ -64,23 +64,21 @@ def __init__(self, observer=MovingAverageMinMaxObserver, quant_min=0, quant_max=
 
     @torch.jit.export
     def enable_fake_quant(self, enabled=True):
```
these can actually just be global functions instead of residing in the module. maybe create an issue to track this?

supriyar replied:
Filed https://github.com/pytorch/pytorch/issues/44751

jerryzh168 commented on test/quantization/test_quantize_fx.py:
```diff
@@ -366,6 +367,44 @@ def forward(self, x):
+
+        # symbolically trace
+        model = symbolic_trace(model)
+        model = prepare_fx(model, qconfig_dict)
```
please use prepare_qat_fx here, we want to be more explicit

supriyar committed:
Update on "[quant][qat] Ensure observers and fq modules are scriptable"


Summary:
Ensure fx module is scriptable after calling prepare_qat on it

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_qat_and_script

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23718380](https://our.internmc.facebook.com/intern/diff/D23718380)

[ghstack-poisoned]

supriyar committed:
Update on "[quant][qat] Ensure observers and fq modules are scriptable"


Summary:
Ensure fx module is scriptable after calling prepare_qat on it

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_qat_and_script

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23718380](https://our.internmc.facebook.com/intern/diff/D23718380)

[ghstack-poisoned]

supriyar committed:
Update on "[quant][qat] Ensure observers and fq modules are scriptable"


Summary:
Ensure fx module is scriptable after calling prepare_qat on it

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_qat_and_script

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23718380](https://our.internmc.facebook.com/intern/diff/D23718380)

[ghstack-poisoned]
